gslib/utils/stet_util.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for Split Trust Encryption Tool integration."""
+
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import division
+from __future__ import unicode_literals
+
+import os
+import shutil
+
+from gslib import storage_url
+from gslib.utils import execution_util
+from gslib.utils import temporary_file_util
+
+from boto import config
+
+DEFAULT_STET_CONFIG_PATH = '~/.config/stet.yaml'
+
+
+class StetSubcommandName(object):
+  """Enum class for available STET subcommands."""
+  ENCRYPT = 'encrypt'
+  DECRYPT = 'decrypt'
+
+
+def _get_stet_binary_from_path():
+  """Retrieves STET binary from path if available. Python 2 compatible."""
+  for path_directory in os.getenv('PATH').split(os.path.pathsep):
+    binary_path = os.path.join(path_directory, 'stet')
+    if os.path.exists(binary_path):
+      return binary_path
+
+
+def _stet_transform(subcommand, blob_id, in_file_path, out_file_path, logger):
+  """Runs a STET transform on a file.
+
+  Encrypts for uploads. Decrypts for downloads. Automatically populates
+  flags for the STET binary.
+
+  Args:
+    subcommand (StetSubcommandName): Subcommand to call on STET binary.
+    blob_id (str): Cloud URL that binary uses for validation.
+    in_file_path (str): File to be transformed source.
+    out_file_path (str): Where to write result of transform.
+    logger (logging.Logger): For logging STET binary output.
+
+  Returns:
+    stet_temporary_file_url (StorageUrl): Path to result of STET-transform.
+
+  Raises:
+    OSError: STET binary was not present in gsutil's boto config file.
+  """
+  binary_path = config.get('GSUtil', 'stet_binary_path', None)
+  if not binary_path:
+    binary_path = _get_stet_binary_from_path()
+  if not binary_path:
+    raise KeyError('Could not find STET binary in boto config or PATH.')
+
+  config_path = config.get('GSUtil', 'stet_config_path', None)
```

Comment:
See my command about not using a default and having the user always specify this explicitly. 

However, if you still want to use the default, I think you can just replace the None here with the default path and then you can just do the existence check on `config_path`.

Reply:
Using the default if the user doesn't specify is what the STET team and I agreed on Friday. Also held a poll with our team.

Reply:
I don't see a mention of default config on the poll. But if the STET team believes that their binary will create this file, then it should be fine.

Reply:
Oh my bad